drivers/flash/Kconfig

```diff
@@ -23,6 +23,16 @@ config FLASH_HAS_PAGE_LAYOUT
 	  This option is enabled when the SoC flash driver supports
 	  retrieving the layout of flash memory pages.
 
+config FLASH_DEVICE_REQUIRES_ERASE
+	bool
+	help
+	  Device requires erase before write.
+
+config FLASH_DEVICE_NOT_REQUIRES_ERASE
+	bool
+	help
+	  Device does not require erase.
+
```

Comment:
>Kconfig are also used with FLASH_ERASE_REQUIRED(condition) macro
that allows to optimize out code that would never be executed

I suggest writing human-readable code and having unambiguous Kconfig options. Before a new Kconfig option is introduced, the author also needs to do some soul-searching and feel the pain of maintenance overhead and deprecation.
Also not clear from commit message why FLASH_ERASE_REQUIRED(condition) needs to be exposed at all and cannot be handled by the drivers.

Reply:
Not going to deprecate them in the near feature as we will have to live with both types of devices.
The Kconfigs have been renamed now and the FLASH_ERASE_REQUIRED has been removed.